token/program2/src/instruction.rs

```diff
@@ -187,6 +187,32 @@ pub enum TokenInstruction {
     ///   2. `[]` The account's multisignature owner.
     ///   3. ..3+M '[signer]' M signer accounts.
     CloseAccount,
+
+    /// Transfers tokens from one account to another either directly or via a delegate.  If this
+    /// account is associated with the native mint then equal amounts of SOL and Tokens will be
+    /// transferred to the destination account.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single owner/delegate
+    ///   0. `[]` The mint for this account
+    ///   1. `[writable]` The source account.
+    ///   2. `[writable]` The source account.
+    ///   3. `[writable]` The destination account.
+    ///   4. '[signer]' The source account's owner/delegate.
+    ///
+    ///   * Multisignature owner/delegate
+    ///   0. `[]` The mint for this account
```

Comment:
Could stick the u8 decimals in the account structure rather than the overhead of loading and spending cycles working with the mint

Reply:
There's a couple problems with that:
1. We got strong encouragement today to avoid changing the Account structure if at all possible
2. I actually do want to validate the mint too.  More on this below...

Reply:
For #1 we would stick it at the end

Reply:
Yeah I know.  We could argue for that if we need to.  I don't think it helps in this case though because I do want to verify the mint